client/send-video.cc

```diff
@@ -104,86 +126,105 @@ int main(int argc, char *argv[]) {
     UDPFlaschenTaschen display(ft_socket, display_width, display_height);
     display.SetOffset(off_x, off_y, off_z);
 
-    // Initalizing these to NULL prevents segfaults!
-    AVFormatContext   *pFormatCtx = NULL;
-    int               i, videoStream;
-    AVCodecContext    *pCodecCtxOrig = NULL;
-    AVCodecContext    *pCodecCtx = NULL;
-    AVCodec           *pCodec = NULL;
-    AVFrame           *pFrame = NULL;
-    AVFrame           *pFrameRGB = NULL;
-    AVPacket          packet;
-    int               frameFinished;
-    int               numBytes;
-    uint8_t           *buffer = NULL;
-    struct SwsContext *sws_ctx = NULL;
-
-    const char *movie_file = argv[optind];
-
     // Register all formats and codecs
     av_register_all();
     avformat_network_init();
 
+    signal(SIGTERM, InterruptHandler);
+    signal(SIGINT, InterruptHandler);
+
+    int numberPlayed = 0;
+    for (int imgarg = optind; imgarg < argc && !interrupt_received; ++imgarg) {
+        const char *movie_file = argv[imgarg];
+
+        bool playresult = PlayVideo(movie_file, display, verbose, repeatTimeout);
+        if (playresult)
+            numberPlayed++;
+
+        if (interrupt_received) {
```

Comment:
This stil needs to be _after_ the loop, otherwise you might end up in a situation in which the message is not printed (if Ctr-C happens after this 'if' and before the next for() loop head).

Also, makes it cleaner.